src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -592,3 +591,41 @@ def _convert_output_format(self, price_energy, csv_output, redis_output):
                 } for timeslot, trades in trade_rates.items()
             ]
             redis_output[node.uuid] = deepcopy(csv_output[node.name])
+
+
+class BaselinePeakEnergyStats:
+    def __init__(self):
+        self.baseline_peak_percentage_result = {}
+        self.baseline_peak_percentage_result_redis = {}
+
+    def update(self, area):
+        self.update_exported_imported_energy(area)
+        self.results_to_redis(area)
+
+    def update_exported_imported_energy(self, area):
+        if area.baseline_energy_settings is not None:
+            area.stats.aggregate_exported_imported_energy(area)
+            baseline_import = area.baseline_energy_settings["baseline_peak_energy_import_kWh"]
+            baseline_export = area.baseline_energy_settings["baseline_peak_energy_export_kWh"]
+            if baseline_import > 0:
+                create_subdict_or_update(self.baseline_peak_percentage_result, area.name,
+                                         {"import": {k: v/baseline_import
```

Comment:
Here the v is a time profile value showing the cumulative imported energy per slot whereas as baseline_import also needs to be a profile value

Reply:
No, v is a value of the area.stats.imported_energy profile, baseline_import is a number.

Reply:
Yes you also said the same thing and I think `baseline_import` also needs to be a profile since its a comparison of results with and without grid fee

Reply:
every value `v` of the `area.stats.imported_energy` is divided by the `baseline_import`